diff -ur sk98lin-10.70.7.3.orig/skge.c sk98lin-10.70.7.3/skge.c
--- sk98lin-10.70.7.3.orig/skge.c	2009-07-15 01:14:49.000000000 +0200
+++ sk98lin-10.70.7.3/skge.c	2009-07-15 01:36:41.661213959 +0200
@@ -571,8 +571,10 @@
 			if (!pSkRootDir) {
 				printk(KERN_WARNING "%s: Unable to create /proc/net/%s",
 					dev->name, SK_Root_Dir_entry);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
 			} else {
 				pSkRootDir->owner = THIS_MODULE;
+#endif
 			}
 		}
 	}
@@ -2327,7 +2329,7 @@
 	
 #ifdef CONFIG_SK98LIN_NAPI
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,23)
-	if (netif_rx_schedule_prep(dev, &pNet->napi)) {
+	if (napi_schedule_prep(&pNet->napi)) {
 #else
 	if (netif_rx_schedule_prep(dev)) {
 #endif
@@ -2336,7 +2338,7 @@
 		pAC->GIni.GIValIrqMask &= ~(NAPI_DRV_IRQS);
 		SK_OUT32(pAC->IoBase, B0_IMSK, pAC->GIni.GIValIrqMask);
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,23)
-		__netif_rx_schedule(dev, &pNet->napi);
+		__napi_schedule(&pNet->napi);
 #else
 		__netif_rx_schedule(dev);
 #endif
@@ -3129,7 +3131,7 @@
 	if(WorkDone < WorkToDo) {
 		spin_lock_irqsave(&pAC->SlowPathLock, Flags);
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,23)
-		__netif_rx_complete(dev, napi);
+		__napi_complete(napi);
 #else
 		netif_rx_complete(dev);
 #endif
