diff -ur sk98lin-10.50.1.3.orig/h/skdrv2nd.h sk98lin-10.50.1.3/h/skdrv2nd.h
--- sk98lin-10.50.1.3.orig/h/skdrv2nd.h	2007-10-22 15:56:58.000000000 +0200
+++ sk98lin-10.50.1.3/h/skdrv2nd.h	2008-05-08 01:06:45.000000000 +0200
@@ -59,22 +59,12 @@
 #include "h/skgeasfconv.h"
 #endif
 
-/* Defines for the poll cotroller */
-#if !defined(SK_NETDUMP_POLL)
-#define SK_NETDUMP_POLL
+#if defined(SK_POLL_CONTROLLER)
+#undef SK_POLL_CONTROLLER
 #endif
-
-#if defined(SK_NETDUMP_POLL)
-#if defined(HAVE_POLL_CONTROLLER)
-#define SK_POLL_CONTROLLER
-#define CONFIG_SK98LIN_NAPI
-#elif defined(CONFIG_NET_POLL_CONTROLLER)
-#define SK_POLL_CONTROLLER
-#if !defined(CONFIG_SK98LIN_NAPI)
+#if defined(CONFIG_SK98LIN_NAPI)
 #define CONFIG_SK98LIN_NAPI
 #endif
-#endif
-#endif
 
 /******************************************************************************
  *
