diff -urN sk98lin-10.70.7.3.orig/skethtool.c sk98lin-10.70.7.3/skethtool.c
--- sk98lin-10.70.7.3.orig/skethtool.c	2009-09-25 23:25:32.000000000 +0200
+++ sk98lin-10.70.7.3/skethtool.c	2009-09-25 23:49:59.722334241 +0200
@@ -1000,7 +1000,11 @@
 	if (netif_running(pAC->dev[port])) {
 		boardWasDown[0] = SK_FALSE;
 	} else {
+#ifdef HAVE_NET_DEVICE_OPS
+		(*pDev->netdev_ops->ndo_open)(pDev);
+#else
 		(*pDev->open)(pDev);
+#endif
 		boardWasDown[0] = SK_TRUE;
 	}
 
@@ -1008,7 +1012,11 @@
 		if (netif_running(pAC->dev[OtherPort])) {
 			boardWasDown[1] = SK_FALSE;
 		} else {
+#ifdef HAVE_NET_DEVICE_OPS
+			(*pOtherDev->netdev_ops->ndo_open)(pOtherDev);
+#else
 			(*pOtherDev->open)(pOtherDev);
+#endif
 			boardWasDown[1] = SK_TRUE;
 		}
 	}
@@ -1085,7 +1093,11 @@
 	     * The board is already up as we bring it up in case it is not.
 	     */
 	  } else {
+#ifdef HAVE_NET_DEVICE_OPS
+	    (*pDev->netdev_ops->ndo_stop)(pDev);
+#else
 	    (*pDev->stop)(pDev);
+#endif
 	  }
 	  if (isDualNetCard) {
 	    if (!boardWasDown[1]) {
@@ -1093,7 +1105,11 @@
 	       * The board is already up as we bring it up in case it is not.
 	       */
 	    } else {
+#ifdef HAVE_NET_DEVICE_OPS
+	      (*pOtherDev->netdev_ops->ndo_stop)(pOtherDev);
+#else
 	      (*pOtherDev->stop)(pOtherDev);
+#endif
 	    }
 	    
 	  }
diff -urN sk98lin-10.70.7.3.orig/skge.c sk98lin-10.70.7.3/skge.c
--- sk98lin-10.70.7.3.orig/skge.c	2009-09-25 23:25:35.000000000 +0200
+++ sk98lin-10.70.7.3/skge.c	2009-09-26 00:11:21.762451012 +0200
@@ -340,6 +340,20 @@
 #endif
 };
 
+#ifdef HAVE_NET_DEVICE_OPS                                                
+static const struct net_device_ops SkGeNetDevOps = {
+	.ndo_open		=  &SkGeOpen,
+	.ndo_stop		=  &SkGeClose,
+	.ndo_get_stats		=  &SkGeStats,
+	.ndo_set_multicast_list	=  &SkGeSetRxMode,
+	.ndo_set_mac_address	=  &SkGeSetMacAddr,
+	.ndo_do_ioctl		=  &SkGeIoctl,
+	.ndo_change_mtu		=  &SkGeChangeMtu,
+#ifdef SK_POLL_CONTROLLER
+	.ndo_poll_controller	=  &SkGeNetPoll,
+#endif
+};
+#endif
 
 /*****************************************************************************
  *
@@ -366,6 +380,7 @@
 	struct SK_NET_DEVICE *dev = NULL;
 	int			retval;
 	int			pci_using_dac;
+	struct net_device_ops	*tmp_ops;
 
 	retval = pci_enable_device(pdev);
 	if (retval) {
@@ -441,6 +456,9 @@
 #endif
 	SET_ETHTOOL_OPS(dev, &sk98lin_ethtool_ops);
 
+#ifdef HAVE_NET_DEVICE_OPS
+	dev->netdev_ops		= &SkGeNetDevOps;
+#else
 	dev->open		=  &SkGeOpen;
 	dev->stop		=  &SkGeClose;
 	dev->get_stats		=  &SkGeStats;
@@ -448,10 +466,11 @@
 	dev->set_mac_address	=  &SkGeSetMacAddr;
 	dev->do_ioctl		=  &SkGeIoctl;
 	dev->change_mtu		=  &SkGeChangeMtu;
-	dev->flags		&= ~IFF_RUNNING;
 #ifdef SK_POLL_CONTROLLER
 	dev->poll_controller	=  SkGeNetPoll;
 #endif
+#endif
+	dev->flags		&= ~IFF_RUNNING;
 	SET_NETDEV_DEV(dev, &pdev->dev);
 
 	pAC->Index = sk98lin_boards_found;
@@ -468,7 +487,12 @@
 
 	/* shifter to later moment in time... */
 	if (CHIP_ID_YUKON_2(pAC)) {
+#ifdef HAVE_NET_DEVICE_OPS
+		tmp_ops = (struct net_device_ops *) dev->netdev_ops;
+		tmp_ops->ndo_start_xmit =	&SkY2Xmit;
+#else
 		dev->hard_start_xmit =	&SkY2Xmit;
+#endif
 #ifdef CONFIG_SK98LIN_NAPI
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,23)
 		netif_napi_add(dev, &pNet->napi, SkY2Poll, 64);
@@ -478,7 +502,12 @@
 #endif
 #endif
 	} else {
+#ifdef HAVE_NET_DEVICE_OPS
+		tmp_ops = (struct net_device_ops *) dev->netdev_ops;
+		tmp_ops->ndo_start_xmit =	&SkGeXmit;
+#else
 		dev->hard_start_xmit =	&SkGeXmit;
+#endif
 #ifdef CONFIG_SK98LIN_NAPI
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,23)
 		netif_napi_add(dev, &pNet->napi, SkGePoll, 64);
@@ -609,7 +638,12 @@
 		pNet->pAC     = pAC;
 
 		if (CHIP_ID_YUKON_2(pAC)) {
+#ifdef HAVE_NET_DEVICE_OPS
+			tmp_ops = (struct net_device_ops *) dev->netdev_ops;
+			tmp_ops->ndo_start_xmit =	&SkY2Xmit;
+#else
 			dev->hard_start_xmit = &SkY2Xmit;
+#endif
 #ifdef CONFIG_SK98LIN_NAPI
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
 			dev->poll =  &SkY2Poll;
@@ -617,7 +651,12 @@
 #endif
 #endif
 		} else {
+#ifdef HAVE_NET_DEVICE_OPS
+			tmp_ops = (struct net_device_ops *) dev->netdev_ops;
+			tmp_ops->ndo_start_xmit =	&SkGeXmit;
+#else
 			dev->hard_start_xmit = &SkGeXmit;
+#endif
 #ifdef CONFIG_SK98LIN_NAPI
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
 			dev->poll =  &SkGePoll;
@@ -628,6 +667,9 @@
 
 		SET_ETHTOOL_OPS(dev, &sk98lin_ethtool_ops);
 
+#ifdef HAVE_NET_DEVICE_OPS
+		dev->netdev_ops		= &SkGeNetDevOps;
+#else
 		dev->open               = &SkGeOpen;
 		dev->stop               = &SkGeClose;
 		dev->get_stats          = &SkGeStats;
@@ -635,10 +677,11 @@
 		dev->set_mac_address    = &SkGeSetMacAddr;
 		dev->do_ioctl           = &SkGeIoctl;
 		dev->change_mtu         = &SkGeChangeMtu;
-		dev->flags             &= ~IFF_RUNNING;
 #ifdef SK_POLL_CONTROLLER
 		dev->poll_controller	= SkGeNetPoll;
 #endif
+#endif
+		dev->flags             &= ~IFF_RUNNING;
 
 #ifdef NETIF_F_TSO
 #ifdef USE_SK_TSO_FEATURE
